src/convertToObject.js

```diff
@@ -6,7 +6,16 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const resultObject = {};
+  const resultArray = sourceString.split(';');
+
+  for (const element of resultArray) {
+    const [key, value] = element.split(':').map((part) => part.trim());
+
+    resultObject[key] = value;
+  }
```

Comment:
The loop assumes that each element in `resultArray` will split into exactly two parts (key and value). If `element` does not contain a colon or is an empty string, this will result in `key` being undefined and `value` being undefined, which might not be the desired behavior. Consider adding a check to ensure that both `key` and `value` are defined before adding them to `resultObject`.